Sem.03/task02.cpp

```diff
@@ -0,0 +1,285 @@
+
+		while (*str != '\0')
+		{
+			count++;
+			str++;
+		}
+
+		return count;
+	}
+
+	void myStrCopy(const char* source, char* dest)
+	{
+		if (source == nullptr || dest == nullptr)
+			return;
+
+		while (*source != '\0')
+		{
+			*dest = *source;
+			dest++;
+			source++;
+		}
+		*dest = '\0';
+	}
+}
+
+struct JobOffer
+{
+	char companyName[MAX_COMPANY_NAME_LEN + 1];
+	int coworkersCount;
+	int paidPaysOff;
+	long long projectSalary;
+};
+
+void initJobOffer(JobOffer& offer,const char* name, int coworkers, int daysOff, long long salary)
+{
+	myFunctions::myStrCopy(name, offer.companyName);
+	offer.coworkersCount = coworkers;
+	offer.paidPaysOff = daysOff;
+	offer.projectSalary = salary;
+}
+
+void saveToFile(const JobOffer& offer, ofstream& ofs)
+{
+	ofs.write((const char*)&offer, sizeof(offer));
+}
+
+void clearBinaryFile(const char* filePath) 
+{
+	ofstream ofs(filePath, ios::out | ios::binary | ios::trunc);
+	ofs.close();
+}
+
+size_t getFileSize(ifstream& ifs)
+{
+	size_t currPos = ifs.tellg();
+	ifs.seekg(0, ios::end);
+	size_t size = ifs.tellg();
+	ifs.seekg(currPos, ios::beg);
+
+	return size;
+}
+
+void readFromFile(ifstream& in, JobOffer& jobOffer)
+{
+	in.read((char*)&jobOffer, sizeof(jobOffer));
+}
+
+void filterOffers(const char* fileName, long long minSalary)
+{
+	JobOffer jobOffer = {};
+	ifstream in(fileName);
+
+	if (!in.is_open())
+	{
+		cout << ERROR_MS << endl;
+		return;
+	}
+
+	size_t fileSize = getFileSize(in);
+	size_t numberOfOrders = fileSize / sizeof(JobOffer);
+
+	for (size_t i = 0; i < numberOfOrders; i++)
+	{
+		in.read((char*)&jobOffer, sizeof(JobOffer));
+		if (jobOffer.projectSalary >= minSalary)
+		{
+			cout << "Company: " << jobOffer.companyName << " covers the minimum" << endl;
+		}
+	}
+	in.close();
+	
+}
+
+bool existOffer(const char* fileName, const char* companyName)
+{
+	JobOffer jobOffer;
+	ifstream in(fileName);
+
+	if (!in.is_open())
+	{
+		cout << ERROR_MS << endl;
+		return false;
+	}
+
+	size_t fileSize = getFileSize(in);
+	size_t numberOfOrders = fileSize / sizeof(JobOffer);
+
+	for (size_t i = 0; i < numberOfOrders; i++)
+	{
+		in.read((char*)&jobOffer, sizeof(JobOffer));
+		if (myFunctions::strCmp(jobOffer.companyName, companyName))
+		{
+			in.close();
+			return true;
+		}
+	}
+	in.close();
+	return false;
+}
+//Bonus 1
+void perfectOffer(const char* filePath, int maxCoworkers, int minVacancyDays, int minSalary) {
+	ifstream fileIn(filePath, ios::binary);
+
+	if (!fileIn.is_open()) {
+		cout << ERROR_MS << endl;
+		return;
+	}
+
+	ofstream fileOut(PERFECT_OFFERS, ios::binary);
+
+	if (!fileOut.is_open()) {
+		cout  << ERROR_MS << endl;
+		return;
+	}
+
+	JobOffer offer = {};
+
+	while (fileIn.read(reinterpret_cast<char*>(&offer), sizeof(JobOffer))) {
+		if (offer.coworkersCount == maxCoworkers && offer.paidPaysOff == minVacancyDays && offer.projectSalary == minSalary) {
+			fileOut.write((const char*)(&offer), sizeof(JobOffer));
+		}
+	}
+
+	fileIn.close();
+	fileOut.close();
+}
+//Bonus2
```

Comment:
implement the system that allows the user to enter different commands. Currently you have hard-coded calls to the functions which are going to be in this system